File: types/agent.ts
/**
 * Shared AgentAction type for AutoWeb monorepo.
 * Used by platform dashboard, project-flow, and backend webhook consumers.
 */

export type AgentStatus =
  | 'detecting'
  | 'analyzing'
  | 'generating'
  | 'deploying'
  | 'escalating';

export type AgentActionType = 'DIRECT_CODE' | 'SLACK_MESSAGE';

export interface AgentAction {
  id: string;
  timestamp: string; // ISO 8601
  status: AgentStatus;
  actionType: AgentActionType;
  payload: string;
  reasoning: string;
}

File: backend/app/schemas/agent.py
"""
Mirrors shared types/agent.ts for webhook payload validation.
"""
from enum import Enum
from typing import Literal, Optional

from pydantic import BaseModel, Field


class AgentStatus(str, Enum):
    DETECTING = "detecting"
    ANALYZING = "analyzing"
    GENERATING = "generating"
    DEPLOYING = "deploying"
    ESCALATING = "escalating"


class AgentActionType(str, Enum):
    DIRECT_CODE = "DIRECT_CODE"
    SLACK_MESSAGE = "SLACK_MESSAGE"


AgentStatusLiteral = Literal["detecting", "analyzing", "generating", "deploying", "escalating"]


class AgentActionPayload(BaseModel):
    """Payload accepted by /agent/webhook (site change from agent)."""

    id: str
    timestamp: Optional[str] = None  # ISO 8601; defaulted in pipeline if missing
    status: Optional[AgentStatusLiteral] = None  # defaulted in pipeline if missing
    actionType: Literal["DIRECT_CODE", "SLACK_MESSAGE"]
    payload: str
    reasoning: str = Field(..., description="Agent reasoning for this action")
